titan-server/src/routes.rs

```diff
@@ -83,4 +87,73 @@ pub fn woltlab_login(
 #[get("/pulse", format = "application/json")]
 pub fn health_check(token: auth_guard::AuthenticatedUser) -> Json<auth_guard::AuthenticatedUser> {
     return Json(token)
+}
+
+/** **************************************************
+ *  Roster
+ ** **************************************************/
+#[derive(Serialize)]
+pub struct RosterMember {
+    user: models::WcfUser,
+    avatar: String
+}
+
+#[derive(Serialize)]
+pub struct FindBranchResponse {
```

Comment:
I see some value in keeping the request/response data guards with the route implementation. Some routes have very specific guards that will likely not be reused. Though, it would still be advantageous to have a scoped `request` module for common pieces.

```
// Located in request module
pub struct Pagination {
    prev: String,
    next: String,
    count: i32,
    page: i32
}
```

```
// Located in roster module
pub struct RosterUsersResponse {
    pub users: Vec<RosterMember>,
    pagination: Pagination
}
```

Reply:
I think if we keep the handler modules small, that can make sense. For the more generic ones, yes, let's move it to a `request` module.

For the future, same thing with anything that might modify a response. Let's make sure they go into a `response` module.